glmnet/logistic.py

```diff
@@ -266,6 +266,17 @@ def _fit(self, X, y, sample_weight=None, relative_penalties=None):
             # keep the original order.
             _y = (y[:, None] == self.classes_).astype(np.float64, order='F')
 
+        # use sample weights, making sure all weights are positive
+        # this is inspired by the R wrapper for glmnet, in lognet.R
+        if sample_weight is not None:
+            weight_gt_0 = sample_weight > 0
+            if not np.all(weight_gt_0):
```

Comment:
The three lines after it should work when all weights are positive. I have a slight preference for avoiding conditionals when possible.